common/k8stest/util_postgres_app.go

```diff
@@ -0,0 +1,314 @@
+			ready = ss.Status.ReadyReplicas == ss.Status.Replicas &&
+				ss.Status.AvailableReplicas == ss.Status.Replicas
+			logf.Log.Info("StatefulSet",
+				"app", "Postgres",
+				"ready", ready,
+				"name", ss.Name,
+				"availableReplicas", ss.Status.AvailableReplicas,
+				"readyReplicas", ss.Status.ReadyReplicas,
+				"currentReplicas", ss.Status.CurrentReplicas,
+			)
+
+		}
+		for !ready && counter > 0 {
+			pods, err := ListPod(psql.Namespace)
+			if err != nil {
+				return err
+			}
+			for _, pod := range pods.Items {
+				if strings.Contains(pod.Name, psql.ReleaseName) {
+					if pod.Status.Phase == coreV1.PodRunning {
+						ready = true
+						psql.Pod = pod
+						logf.Log.Info("Pod",
+							"app", "Postgres",
+							"ready", ready,
+							"name", pod.Name,
+							"status", pod.Status.Phase,
+						)
+						break
+					}
+					logf.Log.Info("Pod",
+						"app", "Postgres",
+						"ready", ready,
+						"name", pod.Name,
+						"status", pod.Status.Phase,
+					)
+				}
+			}
+			if ready {
+				var pvcName = psql.PvcName
+				logf.Log.Info("Verify volume provision", "pvc name", pvcName, "namespace", psql.Namespace)
+				uuid, err := VerifyVolumeProvision(pvcName, psql.Namespace)
+				if err != nil {
+					return fmt.Errorf("failed to verify volume provisioning")
+				}
+				psql.VolUuid = uuid
+				logf.Log.Info("postgres standalone installation is ready")
+				return nil
+			} else {
+				logf.Log.Info("not all apps are ready yet")
+				time.Sleep(10 * time.Second)
+				counter--
+				continue
+			}
+		}
+	}
+	return nil
+}
+
+// Default benchmark parameters
+const (
+	defaultConcurrentDbClients = 4
+	defaultDurationSeconds     = 60
+)
+
+// PgBenchApp represents a PgBench application configuration for benchmarking PostgreSQL.
+type PgBenchApp struct {
+	BenchmarkParams PgBenchmarkParams // Parameters for benchmarking.
+	Name            string            // Name of the PgBench application.
+	Namespace       string            // Kubernetes namespace for the PgBench application.
+	NodeSelector    string            // Node selector for scheduling the PgBench Pod.
+	Pod             *coreV1.Pod       // Kubernetes Pod running the PgBench application.
+}
+
+type PgBenchmarkParams struct {
+	ConcurrentClients int // Number of concurrent clients for PgBench.
+	ThreadCount       int // Number of threads for PgBench.
+	DurationSeconds   int // Duration of the benchmark in seconds.
+}
+
+// NewPgBench creates a new PgBenchApp instance with default benchmark parameters.
+func NewPgBench() *PgBenchApp {
+	return &PgBenchApp{
+		BenchmarkParams: PgBenchmarkParams{
+			ConcurrentClients: defaultConcurrentDbClients,
+			ThreadCount:       defaultThreadCount,
+			DurationSeconds:   defaultDurationSeconds,
+		},
+	}
+}
+
+// InitializePgBench initializes the PgBench database. Must be called before RunPgBench
+func (pgBench *PgBenchApp) InitializePgBench(host string) error {
+	jobName := fmt.Sprintf("%s-init", pgBench.Name)
+	job := &batchV1.Job{
```

Comment:
Can you use builder pattern for job also ?

Reply:
What's resolution of this ?

Reply:
There is a builder for PgBenchApp